suplemon/viewer.py

```diff
@@ -621,3 +604,251 @@ def purge_line_cursors(self, line_no):
         for line_cursors in cursor:
             self.remove_cursor(cursor)
         return True
+
+    ###########################################################################
+    # Input Handling
+    ###########################################################################
+
+    def get_key_bindings(self):
+        """Get list of editor key bindings."""
+        return self.app.get_key_bindings()
+
+    def handle_input(self, event):
+        """Handle input."""
+        if event.type == "mouse":
+            return False
+        key = event.key_code
+        name = event.key_name
+        # Try match a key to a method and call it
+
+        key_bindings = self.get_key_bindings()
+        operation = None
+        if key in key_bindings.keys():
+            operation = key_bindings[key]
+        elif name in key_bindings.keys():
+            operation = key_bindings[name]
+        if operation:
+            self.run_operation(operation)
+            return True
+        return False
+
+    def run_operation(self, operation):
+        """Run an editor core operation."""
+        if operation in self.operations.keys():
+            cancel = self.app.trigger_event_before(operation)
+            if cancel:
+                return False
+            result = self.operations[operation]()
+            self.app.trigger_event_after(operation)
+            return result
+        return False
+
+    ###########################################################################
+    # Operations
+    ###########################################################################
+
+    def arrow_right(self):
+        """Move cursors right."""
+        for cursor in self.cursors:
+            line = self.lines[cursor.y]
+            if cursor.y != len(self.lines)-1 and (cursor.x >= len(line) or len(line) == 0):
```

Comment:
In fact, we can prob move the `cursor.y` check inside of this `if` block since we don't want an else scenario. Then, we can prob convert the `elif` to an `else`